src/middleware/routeHandler.ts

```diff
@@ -108,16 +108,19 @@ export const attachWriteRouteHandlerWrapper: RouteWrapper<{
     return
   }
 
-  Promise.resolve(routeHandler(req, res, next)).catch(async (err: Error) => {
-    await unlock(siteName)
-    next(err)
-  })
-
-  try {
-    await unlock(siteName)
-  } catch (err) {
-    next(err)
-  }
+  Promise.resolve(routeHandler(req, res, next)).then(
+    async () => {
+      try {
+        await unlock(siteName)
+      } catch (err) {
+        next(err)
```

Comment:
Per PR description: does calling `next(err)` make sense here? The route handling was successful and it may have called `next()` itself.